openSUSEway.spec

```diff
@@ -0,0 +1,244 @@
+
+#BRAND: /etc/sway/config and /etc/sway/config.d/
+#BRAND: contain openSUSE config and branding
+
+%description -n sway-branding-openSUSE
+This package provides the openSUSE look and feel for sway.
+
+%package -n     waybar-branding-openSUSE
+Summary:        openSUSE branding of waybar
+Group:          System/GUI/Other
+BuildRequires:  waybar
+Provides:       waybar-branding = %{version}
+Conflicts:      otherproviders(waybar-branding)
+Supplements:    packageand(waybar:branding-openSUSE)
+
+#BRAND: /etc/xdg/waybar/config and /etc/xdg/waybar/style.css
+#BRAND: contain openSUSE config and branding
+
+%description -n waybar-branding-openSUSE
+This package provides the openSUSE look and feel for waybar.
+
+%prep
+%autosetup -p1 -n openSUSEway-%{version}
+
+%build
+
+%install
+
+## openSUSEway package
+### qt5ct config to configure dark theme
+install -D -p -m 644 qt5ct.conf %{buildroot}%{_sysconfdir}/xdg/qt5ct/qt5ct.conf
+### greetd as a login manager
+install -D -p -m 644 greetd/sway-config %{buildroot}%{_sysconfdir}/greetd/sway-config
+install -D -p -m 644 greetd/config.toml %{buildroot}%{_sysconfdir}/greetd/config.toml.way
+install -D -p -m 644 greetd/environments %{buildroot}%{_sysconfdir}/greetd/environments
+
+## openSUSEway pattern package
+mkdir -p %{buildroot}/%{_defaultdocdir}/patterns/
+echo 'This file marks the pattern openSUSEway to be installed.' >%{buildroot}%{_defaultdocdir}/patterns/openSUSEway.txt
+
+## Sway
+install -D -p -m 644 .config/sway/config %{buildroot}%{_sysconfdir}/sway/config
+install -D -p -m 644 .config/sway/env %{buildroot}%{_sysconfdir}/sway/env
+install -D -p -m 644 .config/sway/config.d/50-openSUSE.conf %{buildroot}%{_sysconfdir}/sway/config.d/50-openSUSE.conf
+
+install -D -p -m 644 sway/sway-session.target %{buildroot}%{_unitdir}/sway-session.target
+install -D -p -m 644 sway/sway.service %{buildroot}%{_unitdir}/sway.service
+install -D -p -m 644 sway/sway.desktop %{buildroot}%{_datadir}/wayland-sessions/sway.desktop.brand
+install -D -p -m 755 sway/sway-run.sh %{buildroot}%{_bindir}/sway-run.sh
+
+### alacritty
+# so far doesn't have special branding package and it doesn't support system wide config
+install -D -p -m 644 .config/alacritty/alacritty.yml %{buildroot}%{_sysconfdir}/alacritty/alacritty.yml
+
+## wofi
+install -D -p -m 644 .config/wofi/config %{buildroot}%{_sysconfdir}/wofi/config
+install -D -p -m 644 .config/wofi/style.css %{buildroot}%{_sysconfdir}/wofi/style.css
+#set wofi config and style to the system dir
+sed -i -e "s|wofi --show.*|wofi --conf=%{_sysconfdir}/wofi/config --style=%{_sysconfdir}/wofi/style.css|g" %{buildroot}%{_sysconfdir}/sway/config.d/50-openSUSE.conf
+
+## waybar
+install -D -p -m 644 .config/waybar/config %{buildroot}%{_sysconfdir}/xdg/waybar/config
+install -D -p -m 644 .config/waybar/style.css %{buildroot}%{_sysconfdir}/xdg/waybar/style.css
+
+## wob
+install -D -p -m 644 .config/wob/wob.ini %{buildroot}%{_sysconfdir}/sway/wob/wob.ini
```

Comment:
Installing configuration files under `/etc/sway/$prog` like this means that they will not be autodected by their respective program, meaning that we will have to specify the path to the configuration file in our openSUSEway system configuration file, meaning that the user will not be able to override it.

If we installed (or symlinked) these config files in their default path, the user would be able to override them by dropping the corresponding configuration file in their home.

Reply:
> Installing configuration files under `/etc/sway/$prog` like this means that they will not be autodected by their respective program, meaning that we will have to specify the path to the configuration file in our openSUSEway system configuration file, meaning that the user will not be able to override it.

I think it is just other way around. I would suggest a campaign of upstream submissions to make those programs recognized existence of the system-wide configuration.

Reply:
You mean having 3 levels of autodetected configuration, each one overriding the previous?
```
system configuration: /etc/$prog
                          ^
branding configuration: something like /etc/branding/$prog or /usr/share/branding/$prog
                          ^
user configuration: ~/.config/$prog or $XDG_CONFIG_HOME/$prog
```

Reply:
> You mean having 3 levels of autodetected configuration, each one overriding the previous?
> 
> ```
> system configuration: /etc/$prog
>                           ^
> branding configuration: something like /etc/branding/$prog or /usr/share/branding/$prog
>                           ^
> user configuration: ~/.config/$prog or $XDG_CONFIG_HOME/$prog
> ```

I think, if we have what you call “branding configuration” then the system one is not needed, is it?

Reply:
I don't know, isn't $prog going to overwrite the configuration in /etc/$prog everytime it's updated anyway?

Reply:
> I don't know, isn't $prog going to overwrite the configuration in /etc/$prog everytime it's updated anyway?

It must be marked as `%config`, yes. https://docs.fedoraproject.org/en-US/packaging-guidelines/#_configuration_files

Reply:
There is also:
https://en.opensuse.org/openSUSE:Packaging_UsrEtc

so host specific is `/etc/` , distribution-provided  is `/usr/etc/` , and user specific `~/.config/` .

But that doesn't solve the **branding** problem, which is neither of those. It is probably more related to the `config.d/*` style of configs, so both default and branding could co-exist and override each other by prio.

Or probably distribution-provided is a default and host-specific is a branding.